app/uk/gov/hmrc/gform/eval/EvaluationResults.scala

```diff
@@ -238,7 +238,12 @@ case class EvaluationResults(
   def evalTyped(er: ExpressionResult, tpe: ExplicitExprType): ExpressionResult =
     tpe match {
       case ExplicitExprType.Sterling(roundingMode) =>
-        er.withNumberResult(bigDecimal => NumberSetScale.setScale(bigDecimal, 2, roundingMode))
+        er match {
+          case NumberResult(_) =>
+            er.withNumberResult(bigDecimal => NumberSetScale.setScale(bigDecimal, 2, roundingMode))
+          case _ => er
+        }
```

Comment:
How is this not the same as 
```
er.withNumberResult(bigDecimal => NumberSetScale.setScale(bigDecimal, 2, roundingMode))
```
? 

Can you provide minimal example where this makes a difference?

Reply:
I think we should round only `NumberResult`, not `ListResult`

Reply:
Reverted